crates/bevy_ecs/src/query/iter.rs

```diff
@@ -1203,6 +1250,33 @@ impl<'w, 's, D: ReadOnlyQueryData, F: QueryFilter, I: Iterator<Item: Borrow<Enti
     }
 }
 
+impl<
+        'w,
+        's,
+        D: ReadOnlyQueryData,
+        F: QueryFilter,
+        I: DoubleEndedIterator<Item: Borrow<Entity>>,
+    > DoubleEndedIterator for QueryManyIter<'w, 's, D, F, I>
+{
+    #[inline(always)]
+    fn next_back(&mut self) -> Option<Self::Item> {
+        // SAFETY:
+        // All arguments stem from self.
+        // It is safe to alias for ReadOnlyWorldQuery.
+        unsafe {
+            Self::fetch_next_aliased_unchecked(
+                self.entity_iter.by_ref().rev().by_ref(),
```

Comment:
Do you need the second `by_ref()` here?  You only have one in `fetch_next_back()`, and that should be enough to prevent `self.entity_iter` from being moved.

Reply:
you're right, I think I missed this in my second commit!

Reply:
removed.